### ISSUE TITLE
Convolution2D input error

### ISSUE THREAD

**Issue by mramz** [fchollet/keras#3852]
I having trouble with the input for the convolution2D net. I have 2 samples of dimension 1x12 and a convolution layer with 5 filters and kernel of 1x3. My data looks like

```
data=np.array([[ 0.06598175,  0.12817432,  0.36949296,  0.5712227 , 
         0.54105666, 0.38620151,  0.20075556,  0.11366828,  0.07925526,  
         0.05270926, 0.08046888,  0.07579738],
       [ 0.06797359,  0.13533782,  0.37340452,  0.57014065,  0.53585215,
        0.39121396,  0.20627117,  0.10934431,  0.07212514,  0.04386237,
        0.07302206,  0.07364738]])
```

i get the error:  
Error when checking model target: expected activation_2 to have shape (None, 5, 1, 10) but got array with shape (2, 1, 1, 12)

Why is it expecting a different input shape when i specified it as input_shape=(1, 1, 12)

```
from keras.layers import Flatten,Dropout, Dense,Activation,Convolution2D,MaxPooling2D
from keras.models import Sequential
import numpy as np

dataT = data.reshape(data.shape[0], 1, 1, 12)

model = Sequential()

model.add(Convolution2D(5, 1, 3, border_mode='valid', input_shape=(1, 1, 12)))
model.add(Activation('relu'))

model.compile(loss='mse', optimizer='adam')

model.fit(dataT, dataT, batch_size=2, nb_epoch=10)
```

**Comment by robertomest** [comment]
The problem is actually with the target. You are trying to use dataT as input and targets, but the output of your network is actually (None, 5, 1, 10) because you used 5 filters and no padding. You should either change the convolutional layer to have the 1 filter and border_mode='same' or change the targets in order to have compatible sizes.

**Comment by mramz** [comment]
Ahh ok thanks my that make a ton of sense new to keras so thanks for your help. my full model is

```
model = Sequential()
# input: 100x100 images with 3 channels -> (3, 100, 100) tensors.
# this applies 32 convolution filters of size 3x3 each.
model.add(Convolution2D(5, 1, 3, border_mode='valid', input_shape=(1, 1, 12)))
model.add(Convolution2D(12, 1, 3))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(1, 2)))
# model.add(Dropout(0.25))

model.add(Flatten())

model.add(Dense(48))
model.add(Activation('relu'))
# model.add(Dropout(0.5))

model.add(Dense(12))
model.add(Activation('softmax'))


model.compile(loss='mse', optimizer='SGD')
model.fit(dataT, dataT, batch_size=2, nb_epoch=1)
```

so how to i get the right out put dimension for my last layer? Sorry if it a stupid question still learning

**Comment by robertomest** [comment]
This seems to work for me. You can remove the dim_ordering arguments, I just used them to run the code using tensorflow as backend. I used the [Reshape layer](https://keras.io/layers/core/#reshape) to make the (None, 12) output into (None, 1, 1, 12) like you needed.
```
from keras.layers import Flatten,Dropout, Dense,Activation,Convolution2D,MaxPooling2D, Reshape
from keras.models import Sequential
import numpy as np

data=np.array([[ 0.06598175,  0.12817432,  0.36949296,  0.5712227 ,
         0.54105666, 0.38620151,  0.20075556,  0.11366828,  0.07925526,
         0.05270926, 0.08046888,  0.07579738],
       [ 0.06797359,  0.13533782,  0.37340452,  0.57014065,  0.53585215,
        0.39121396,  0.20627117,  0.10934431,  0.07212514,  0.04386237,
        0.07302206,  0.07364738]])

dataT = data.reshape(data.shape[0], 1, 1, 12)

model = Sequential()
# input: 100x100 images with 3 channels -> (3, 100, 100) tensors.
# this applies 32 convolution filters of size 3x3 each.
model.add(Convolution2D(5, 1, 3, border_mode='valid', input_shape=(1, 1, 12), dim_ordering='th'))
model.add(Convolution2D(12, 1, 3, dim_ordering='th'))
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(1, 2), dim_ordering='th'))
# model.add(Dropout(0.25))

model.add(Flatten())

model.add(Dense(48))
model.add(Activation('relu'))
# model.add(Dropout(0.5))

model.add(Dense(12))
model.add(Activation('softmax'))
model.add(Reshape((1, 1, 12)))

model.compile(loss='mse', optimizer='SGD')
model.fit(dataT, dataT, batch_size=2, nb_epoch=1)
```

You might want to define a optimizer instead of just using the default, that will allow you to tune learning rate, momentum and other parameters. You can do it like so

````
from keras.optimizers import SGD
#SGD with some parameters
opt = SGD(lr=0.01, momentum=0.9)
model.compile(loss='mse', optimizer=opt)
```

You can choose any of the [implemented optimizers](https://keras.io/optimizers). You might also want to read the [blog post](https://blog.keras.io/building-autoencoders-in-keras.html) about autoencoders.

**Comment by mramz** [comment]
Cool thanks too much for your help! Really cleared up a lot of my question about how the input output layers work.
